test/ad/distributions.jl

```diff
@@ -34,20 +35,35 @@
         return S, pullback
     end
 
-    # Create matrix `X` such that `X` and `I - X` are positive definite
+    # Create matrix `X` such that `X` and `I - X` are positive definite if `A ≠ 0`.
     function to_beta_mat(A)
         S = A * A' + I
         invL = inv(cholesky(S).L)
         return invL * invL'
     end
 
+    # Create positive values.
+    to_positive(x) = exp.(x)
+    to_positive(x::AbstractArray{<:AbstractArray}) = to_positive.(x)
+
+    # Create vectors in probability simplex.
+    # Custom implementation since `StatsFuns.softmax` is not compatible with Zygote.
+    function to_simplex(x::AbstractArray; dims=1)
+        maxx = maximum(x, dims=dims)
+        y = exp.(x .- maxx)
+        return y ./ sum(y, dims=dims)
+    end
+    to_simplex(x::AbstractArray{<:AbstractArray}; dims=1) = to_simplex.(x; dims=dims)
```

Comment:
We already depend on `NNlib` for this exact reason,  you can just replace this with `NNlib.softmax` :+1:

Reply:
Ah nice :+1:

I didn't realize this when writing the tests in DistributionsAD (and there NNlib is an indirect dependency).

Reply:
BTW it seems Bijectors uses only `NNlib.softplus` but nothing esle from `NNlib`?

Reply:
Yeah but that's for the same reason I believe. Also, the PR for rational quadratic splines (which no one seems to want to look at :sweat_smile: ) uses `NNlib.softmax` for the above reason.